lib/src/actions/actions.dart

```diff
@@ -2729,6 +2730,92 @@ abstract class DNAEndsMoveCommit
   String short_description() => "move DNA ends";
 }
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// dna extensions move
+
+abstract class DNAExtensionsMoveStart
+    with BuiltJsonSerializable
+    implements Action, Built<DNAExtensionsMoveStart, DNAExtensionsMoveStartBuilder> {
+  Point<num> get start_point;
+
+  Helix get helix;
+
+  /************************ begin BuiltValue boilerplate ************************/
+  factory DNAExtensionsMoveStart({Point<num> start_point, Helix helix}) = _$DNAExtensionsMoveStart._;
+
+  DNAExtensionsMoveStart._();
+
+  static Serializer<DNAExtensionsMoveStart> get serializer => _$dNAExtensionsMoveStartSerializer;
+}
+
+abstract class DNAExtensionsMoveSetSelectedExtensions
+    with BuiltJsonSerializable
+    implements
+        Action,
+        Built<DNAExtensionsMoveSetSelectedExtensions, DNAExtensionsMoveSetSelectedExtensionsBuilder> {
+  BuiltList<DNAExtensionMove> get moves;
+  //TODO: finish implementing this
+  Point<num> get original_point;
+
+  BuiltSet<Strand> get strands_affected;
+
+  Helix get helix;
+
+  /************************ begin BuiltValue boilerplate ************************/
+  factory DNAExtensionsMoveSetSelectedExtensions(
```

Comment:
A better name is `DNAExtensionsMoveSetSelectedExtensionEnds` (it's the ends of the extensions being selected, not the extensions themselves, unless I'm misunderstanding something). Same thing with the associated reducer.